### PR TITLE
[enhancement] create test/utils package for sklearnex testing 

### PR DESCRIPTION
**Pull request by icfaust**
### Description

move ```_utils.py``` and ```_utils_spmd.py``` to a separate package to make further modifications/expansions of these utilities easier.

Renames ```_sklearn_clone_dict``` to ```sklearn_clone_dict```
Removes a relative import from test_run_to_run_stability.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Review by samir-nasibli — APPROVED** [review]
I am good with this refactoring. Please rebase and rerun CI again. expecting green CI now
